CondFormats/RPCObjects/test/RPCReadOutMapAnalyzer.cc

```diff
@@ -17,37 +17,35 @@ using namespace std;
 using namespace edm;
 
 // class declaration
-class RPCReadOutMapAnalyzer : public edm::EDAnalyzer {
+class RPCReadOutMapAnalyzer : public edm::one::EDAnalyzer<> {
 public:
   explicit RPCReadOutMapAnalyzer(const edm::ParameterSet&);
-  ~RPCReadOutMapAnalyzer();
-  virtual void analyze(const edm::Event&, const edm::EventSetup&);
+  ~RPCReadOutMapAnalyzer() override = default;
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
 
 private:
   bool m_flag;
+  edm::ESGetToken<RPCEMap, RPCEMapRcd> readoutMappingToken_;
+  edm::ESGetToken<RPCReadOutMapping, RPCReadOutMappingRcd> mapZeroToken_;
 };
 
 RPCReadOutMapAnalyzer::RPCReadOutMapAnalyzer(const edm::ParameterSet& iConfig)
-    : m_flag(iConfig.getUntrackedParameter<bool>("useNewEMap", false)) {
+    : m_flag(iConfig.getUntrackedParameter<bool>("useNewEMap", false)),
+      readoutMappingToken_(esConsumes()),
+      mapZeroToken_(esConsumes()) {
   ::putenv(const_cast<char*>(std::string("CORAL_AUTH_USER konec").c_str()));
   ::putenv(const_cast<char*>(std::string("CORAL_AUTH_PASSWORD konecPass").c_str()));
 }
 
-RPCReadOutMapAnalyzer::~RPCReadOutMapAnalyzer() {}
-
 void RPCReadOutMapAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
   std::cout << "====== RPCReadOutMapAnalyzer" << std::endl;
 
   const RPCReadOutMapping* map;
   if (m_flag) {
-    edm::ESHandle<RPCEMap> readoutMapping;
-    iSetup.get<RPCEMapRcd>().get(readoutMapping);
-    const RPCEMap* eMap = readoutMapping.product();
-    //     RPCReadOutMapping* map = eMap->convert();
-    map = eMap->convert();
+    edm::ESHandle<RPCEMap> readoutMapping = iSetup.getHandle(readoutMappingToken_);
+    map = readoutMapping.product()->convert();
```

Comment:
the Handle is not used for anything so this could be make concise like
```suggestion
    const auto map =  &iSetup.getHandle(readoutMappingToken_)->convert();
```
although I'm not really sure what the convert() is doing

Reply:
I removed the redundant vars but I'm not sure how to switch to getData instead of getHandle

Reply:
I gave an explicit suggestion here
https://github.com/cms-sw/cmssw/pull/36148#discussion_r750806708
HTH